app/assets/javascripts/details/shell/c2.js

```diff
@@ -1,8 +1,8 @@
 var C2;
-
 C2 = (function() {
   
-  function C2(){
+  function C2(config){
+    config = config || {};
```

Comment:
I think this was the cause of the errors. I was using a ES6 JS syntax for assigning default params and it was failing. @nickbristow